pkg/operator/sync.go

```diff
@@ -116,13 +116,18 @@ func (optr *Operator) syncTerminationHandler(config *OperatorConfig) error {
 	return nil
 }
 
-func (optr *Operator) syncBaremetalControllers(config *OperatorConfig) error {
-	// Try to get baremetal provisioning config from a CR
-	baremetalProvisioningConfig, err := getBaremetalProvisioningConfig(optr.dynamicClient, baremetalProvisioningCR)
-	if err != nil {
-		glog.Errorf("Unable to read Baremetal Provisioning config from CR %s.", baremetalProvisioningCR)
-		glog.Infof("Will try to read Baremetal Provisioning config from ConfigMap %s instead", baremetalConfigmap)
+func (optr *Operator) checkIfBaremetalAndGetConfig(config *OperatorConfig) (BaremetalProvisioningConfig, bool) {
```

Comment:
this function is doing too many things and its not reused anywhere. This logic can just be inlined here https://github.com/openshift/machine-api-operator/pull/560/files#diff-fa45321336db7ad1cedc28bf643a4f97R59

Reply:
Done.